server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -321,7 +321,7 @@ public void onComplete(Result result)
                         "query/time",
                         requestTime,
                         "success",
-                        result.isSucceeded()
+                        result.isSucceeded() && result.getResponse().getStatus() == 200
```

Comment:
It was a bug, isSucceeded only returns false when there is an error in processing the async request, in case of timeouts the broker returns a successful response with status set to INTERNAL_SERVER_ERROR i.e 500. 

have modified to use static variable.